tests/positive/partitions/creation-mb.go

```diff
@@ -0,0 +1,330 @@
+			{
+				Label:    "create-partition",
+				Number:   1,
+				Length:   65536,
+				TypeGUID: "B921B045-1DF0-41C3-AF44-4C6F280D3FAE",
+				GUID:     "05AE8178-224E-4744-862A-4F4B042662D0",
+			},
+		},
+	})
+	config := `{
+		"ignition": {
+			"version": "$version"
+		},
+		"storage": {
+			"disks": [
+			{
+				"device": "$disk1",
+				"partitions": [
+				{
+					"number": 1,
+					"size-mb": 32,
+					"label": "create-partition",
+					"typeGuid": "B921B045-1DF0-41C3-AF44-4C6F280D3FAE",
+					"guid": "05AE8178-224E-4744-862A-4F4B042662D0"
+				}
+				]
+			}
+			]
+		}
+	}`
+	return types.Test{
+		Name:             name,
+		In:               in,
+		Out:              out,
+		Config:           config,
+		ConfigMinVersion: "2.3.0-experimental",
+	}
+}
+
+func CreatePartitionMBWithStart() types.Test {
+	name := "Create a single partition on a blank disk using MBs providing the starting sector"
+	in := append(types.GetBaseDisk(), types.Disk{Alignment: types.IgnitionAlignment})
+	out := append(types.GetBaseDisk(), types.Disk{
+		Alignment: types.IgnitionAlignment,
+		Partitions: types.Partitions{
+			{
+				Label:    "create-partition",
+				Number:   1,
+				Length:   65536,
+				TypeGUID: "B921B045-1DF0-41C3-AF44-4C6F280D3FAE",
+				GUID:     "05AE8178-224E-4744-862A-4F4B042662D0",
+			},
+		},
+	})
+	config := `{
+		"ignition": {
+			"version": "$version"
+		},
+		"storage": {
+			"disks": [
+			{
+				"device": "$disk1",
+				"partitions": [
+				{
+					"number": 1,
+					"start-mb": 1,
+					"size-mb": 32,
+					"label": "create-partition",
+					"typeGuid": "B921B045-1DF0-41C3-AF44-4C6F280D3FAE",
+					"guid": "05AE8178-224E-4744-862A-4F4B042662D0"
+				}
+				]
+			}
+			]
+		}
+	}`
+	return types.Test{
+		Name:             name,
+		In:               in,
+		Out:              out,
+		Config:           config,
+		ConfigMinVersion: "2.3.0-experimental",
+	}
+}
+
+func WipeAndCreateNewPartitionsMB() types.Test {
+	name := "Wipe disk and create new partitions using MB"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": {
+			"version": "$version"
+		},
+		"storage": {
+			"disks": [
+			{
+				"device": "$disk1",
+				"wipeTable": true,
+				"partitions": [
+				{
+					"label": "important-data",
+					"number": 1,
+					"size-mb": 32,
+					"typeGuid": "B921B045-1DF0-41C3-AF44-4C6F280D3FAE",
+					"guid": "8A7A6E26-5E8F-4CCA-A654-46215D4696AC"
+				},
+				{
+					"label": "ephemeral-data",
+					"number": 2,
+					"size-mb": 64,
+					"typeGuid": "CA7D7CCB-63ED-4C53-861C-1742536059CC",
+					"guid": "A51034E6-26B3-48DF-BEED-220562AC7AD1"
+				}
+				]
+			}
+			]
+		}
+	}`
+	// Create dummy partitions. The UUIDs in the input partitions
+	// are intentionally different so if Ignition doesn't do the right thing the
+	// validation will fail.
+	in = append(in, types.Disk{
+		Alignment: types.IgnitionAlignment,
+		Partitions: types.Partitions{
+			{
+				Label:    "important-data",
```

Comment:
Might be worthwhile to muck up the input partitions a bit more to ensure that the other pieces are being executed rather than just `GUID`

Reply:
I'm not very concerned about that since the matching logic for the other bits is tested in the sector'd code.